src/Render/Markdown/D2LLinkInlineRenderer.cs

```diff
@@ -78,12 +78,15 @@ LinkInline link
 			if( url.StartsWith( "/" ) ) {
 				rewritten = $"/{context.DocRootRepoName}{rewritten}";
 			}
-
+
+			// Change extension in standard case
 			if ( rewritten.EndsWith( ".md" ) ) {
-				rewritten = rewritten.Substring( 0, rewritten.Length - 3 ) + ".html";
-			}
-
-			return rewritten;
+				return rewritten.Substring( 0, rewritten.Length - 3 ) + ".html";
+			}
+
+			// Change extension if there's an anchor.
+			// We avoid changing all ".md" to ".html", otherwise we could easily mess up other parts of the url.
```

Comment:
nit: Working with a `Uri` object here to update the `Path` and leave the other parts alone may make this simpler.

Reply:
I took a stab at that, but [`Uri` doesn't play well with relative paths](https://docs.microsoft.com/en-us/dotnet/api/system.uri.pathandquery?view=net-5.0#exceptions). `UriBuilder` similarly seems to be designed for creating absolute URIs.